MapboxCoreNavigation/MMEEventsManager.swift

```diff
@@ -0,0 +1,164 @@
+        modifiedEventDictionary["batteryLevel"] = UIDevice.current.batteryLevel >= 0 ? UIDevice.current.batteryLevel * 100 : -1
+        modifiedEventDictionary["applicationState"] = UIApplication.shared.applicationState.telemString
+        
+        //modifiedEventDictionary["connectivity"] = ??
+        
+        return modifiedEventDictionary
+    }
+}
+
+extension UIApplicationState {
+    var telemString: String {
+        get {
+            switch self {
+            case .active:
+                return "Foreground"
+            case .inactive:
+                return "Inactive"
+            case .background:
+                return "Background"
+            }
+        }
+    }
+}
+
+extension UIDevice {
+    var machine: String {
+        get {
+            var systemInfo = utsname()
+            uname(&systemInfo)
+            let machineMirror = Mirror(reflecting: systemInfo.machine)
+            let identifier = machineMirror.children.reduce("") { identifier, element in
+                guard let value = element.value as? Int8, value != 0 else { return identifier }
+                return identifier + String(UnicodeScalar(UInt8(value)))
+            }
+            
+            return identifier
+        }
+    }
+}
+
+extension CLLocation {
+    var dictionary: [String: Any] {
+        get {
+            var locationDictionary:[String: Any] = [:]
+            locationDictionary["lat"] = coordinate.latitude
+            locationDictionary["lng"] = coordinate.longitude
+            locationDictionary["altitude"] = altitude
+            locationDictionary["timestamp"] = timestamp.ISO8601
+            locationDictionary["horizontalAccuracy"] = horizontalAccuracy
+            locationDictionary["verticalAccuracy"] = verticalAccuracy
+            locationDictionary["course"] = course
+            locationDictionary["speed"] = speed
+            return locationDictionary
+        }
+    }
+}
+
+class FixedLengthBuffer<T> {
+    private var objects = Array<T>()
+    private var length: Int
+    
+    public init(length: Int) {
+        self.length = length
+    }
+    
+    public func push(_ obj: T) {
+        objects.append(obj)
+        if objects.count == length {
+            objects.remove(at: 0)
+        }
+    }
+    
+    public var allObjects: Array<T> {
+        get {
+            return Array(objects)
+        }
+    }
+}
+
+class CoreFeedbackEvent: Hashable {
+    var id = UUID()
+    
+    var timestamp: Date
+    
+    var eventDictionary: [String: Any]
+    
+    init(timestamp: Date, eventDictionary: [String: Any]) {
+        self.timestamp = timestamp
+        self.eventDictionary = eventDictionary
+    }
+    
+    var hashValue: Int {
+        get {
+            return id.hashValue
+        }
+    }
+    
+    static func ==(lhs: CoreFeedbackEvent, rhs: CoreFeedbackEvent) -> Bool {
+        return lhs.id == rhs.id
+    }
+}
+
+class FeedbackEvent: CoreFeedbackEvent {}
+
+class RerouteEvent: CoreFeedbackEvent {}
```

Comment:
Per https://github.com/mapbox/mapbox-navigation-ios/pull/304/#discussion_r124867003, it would be great to see this cluster of classes become a single struct containing an enumeration-typed property. (Down the line, if the two kinds of events begin to diverge, we can replace the entire struct with an enumeration with associated values.)